### PR TITLE
Added CLI mode for il_trafficgen.

### PR DESCRIPTION
**Pull request by a-jamshed**
il_trafficgen was previously not adjusted to work in CLI mode. This patch should do the fix.

Signed-off-by: Muhammad Asim Jamshed <muhammad.jamshed@intel.com>

### CODE REVIEW

File: pktgen/app/pktgen-tool.c [hunk 1]
```diff
@@ -418,6 +421,40 @@ void stop_pkt_frwding(int sig __rte_unused) {
   is_first_tx = true;
 }
 
+/* Function: start_gtp_traffic
+ * Input param: None
+ * Output param: None
+ * Return values: None
+ */
+void start_gtp_traffic(int sig __rte_unused) {
```
Comment (ghost) [inline pktgen/app/pktgen-tool.c:429]: it composes the traffic with gtp header or just send normal traffic and go through gtp tunnel?

Comment (a-jamshed) [reply pktgen/app/pktgen-tool.c:429]: Depends on the command line argument (If the command used to start il_trafficgen was '-g', it starts sending tunneled traffic. If the command to start il_trafficgen was '-r', it starts sending normal traffic.) This is the cli version of what you would have done in the tmux screen: "start 0".

**Comment by ghost** [comment]
+1

File: il_nperf_cli.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	s1u_cores="[$((${cores[0]}+1)):$((${cores[0]}+2))].0"
+	cmd="./app/x86_64-native-linuxapp-gcc/pktgen -l $core_list -n 4 --proc-type auto  \
+		--log-level 7 --socket-mem $NUMA0_MEM,$NUMA1_MEM --file-prefix s1u_pg -w $s1u_port -- -t -T     \
+		-N -m $s1u_cores -f ./autotest/input.txt -f themes/black-yellow.theme -g 127.0.0.1:2222"
+elif [[ $1 == "-r" ]] || [[ $1 == "--responder" ]]; then
+	#sgi_cores="[$(( ${cores[0]}+5 ))-$(( ${cores[0]}+6)):$(( ${cores[0]}+7 ))-$(( ${cores[0]}+8))].0"
+	sgi_cores="[$((${cores[0]}+3)):$((${cores[0]}+4))].0"
+	cmd="./app/x86_64-native-linuxapp-gcc/pktgen -l $core_list -n 4 --proc-type auto  \
+		--log-level 7 --socket-mem $NUMA0_MEM,$NUMA1_MEM --file-prefix sgi_pg -w $sgi_port -- -r -T     \
+		-N -m $sgi_cores -f ./autotest/input.txt -f themes/black-yellow.theme -g 127.0.0.1:3333"
```
Comment (somnathchakrabarti) [inline il_nperf_cli.sh:15]: Asim : why do we have to put the -g flag even in responder mode (line#15 in il_nperf_cli.sh) ?

File: pktgen/app/pktgen-tool.c [hunk 3]
```diff
@@ -291,6 +291,9 @@ int parse_user_config(void) {
 
   if (config_lookup_int(&cfg, "enable_gtpu", &val)) {
     enable_gtpu = val;
+    /* register user additional user signals in case pktgen needs to handled via signals */
+    signal(SIGUSR1, start_gtp_traffic);
+    signal(SIGUSR2, quit_gtp_traffic);
```
Comment (somnathchakrabarti) [inline pktgen/app/pktgen-tool.c:296]: This is a nice thought. Please also document this way of controlling the traffic_gen in README with an example

File: README.md [hunk 4]
```diff
@@ -120,6 +121,34 @@ il_trafficgen stops automatically, after test-duration.
...
+	$ socat - TCP4:localhost:2222 < start_il_trafficgen.lua
+
+	4. To send downlink traffic
+	Send the following command to generate Downlink traffic(i.e., P0-Rx <-- P1-Tx) on RESPONDER.
+	$ socat - TCP4:localhost:3333 < start_il_trafficgen.lua
+
+	5. To send traffic uplink and downlink run step 3 and 4 simultaneously.
+
+	6. Send the following command to quit both GENERATOR and RESPONDER.
+	$ socat - TCP4:localhost:2222 < quit_il_trafficgen.lua ; socat - TCP4:localhost:3333 < quit_il_trafficgen.lua
```
Comment (somnathchakrabarti) [inline README.md:152]: "socat - TCP4:localhost:2222 < quit_il_trafficgen.lua" - would be nice to get the traffic statistics returned on port 2222 after stopping the generator. This way we don't have to look at the pktgen screen to see the results.

Comment (a-jamshed) [reply README.md:152]: Traffic statistics are always logged to output files in `log/` subdirectory. We don't really need to look at the pktgen screen. In fact, if the cli mode is used to run il_trafficgen, the user  won't need to see the screen at all.

File: il_nperf_cli.sh (hunk 2 above)
Comment (a-jamshed) [reply il_nperf_cli.sh:15]: This `-g` does not stand for "generator" mode. This switch is part of the original dpdk pktgen interface. Please see [this](https://pktgen-dpdk.readthedocs.io/en/latest/usage_pktgen.html) for more details. This switch is used to let the process create sockets and start listening on arbitrary TCP ports. The contents of the lua script (e.g. `start_il_trafficgen.lua`) are sent to these sockets (and, hence, can be used to control the operations of the processes).

File: pktgen/app/pktgen-tool.c (hunk 3 above)
Comment (a-jamshed) [reply pktgen/app/pktgen-tool.c:296]: Thanks for the reminder. I forgot to document this.

**Pull request merged**